src/libraries/System.Private.Xml/src/System/Xml/Schema/DtdParserAsync.cs

```diff
@@ -1093,11 +1097,11 @@ private async Task ParseCondSectionAsync()
             }
         }
 
-        private async Task<Tuple<string, string>> ParseExternalIdAsync(Token idTokenType, Token declType)
+        private async Task<Tuple<string?, string?>> ParseExternalIdAsync(Token idTokenType, Token declType)
         {
-            Tuple<string, string> tuple;
-            string publicId;
-            string systemId;
+            Tuple<string?, string?> tuple;
+            string? publicId;
+            string? systemId;
```

Comment:
Why is this null? It seems to me that `systemId` gets a value on all the branches that doesn't throw.

Reply:
Ah nvm, `systemId` can be null when all of this conditions met:

`!(idTokenType == Token.SYSTEM)` && 
`!(declType == Token.DOCTYPE && !_freeFloatingDtd)` && 
!`(await GetTokenAsync(false).ConfigureAwait(false) == Token.Literal)` && 
`!(declType != Token.NOTATION)`

Reply:
actually there is an `else` statement missing inside the top level `else` (although I initially couldn't see it). Also it doesn't matter that much since rest of the code is ok with `string?` for both `systemId` and `publicId` and this particular API is private